statsmodels/gam/smooth_basis.py

```diff
@@ -0,0 +1,654 @@
+
+        return
+
+    def _smooth_basis_for_single_variable(self):
+        basis = dmatrix("cc(x, df=" + str(self.df) + ") - 1", {"x": self.x})
+        n_inner_knots = self.df - 2 + 1  # +n_constraints # TODO: from CubicRegressionSplines class
+        all_knots = _get_all_sorted_knots(self.x, n_inner_knots=n_inner_knots, inner_knots=None,
+                                          lower_bound=None, upper_bound=None)
+
+        b, d = self._get_b_and_d(all_knots)
+        s = self._get_s(b, d)
+
+        return basis, None, None, s
+
+    def _get_b_and_d(self, knots):
+        """Returns mapping of cyclic cubic spline values to 2nd derivatives.
+
+        .. note:: See 'Generalized Additive Models', Simon N. Wood, 2006, pp 146-147
+
+        :param knots: The 1-d array knots used for cubic spline parametrization,
+         must be sorted in ascending order.
+        :return: A 2-d array mapping cyclic cubic spline values at
+         knots to second derivatives.
+        """
+        h = knots[1:] - knots[:-1]
+        n = knots.size - 1
+
+        # b and d are defined such that the penalty matrix is equivalent to:
+        # s = d.T.dot(b^-1).dot(d)
+        # reference in particular to pag 146 of Wood's book
+        b = np.zeros((n, n))  # the b matrix on page 146 of Wood's book
+        d = np.zeros((n, n))  # the d matrix on page 146 of Wood's book
+
+        b[0, 0] = (h[n - 1] + h[0]) / 3.
+        b[0, n - 1] = h[n - 1] / 6.
+        b[n - 1, 0] = h[n - 1] / 6.
+
+        d[0, 0] = -1. / h[0] - 1. / h[n - 1]
+        d[0, n - 1] = 1. / h[n - 1]
+        d[n - 1, 0] = 1. / h[n - 1]
+
+        for i in range(1, n):
+            b[i, i] = (h[i - 1] + h[i]) / 3.
+            b[i, i - 1] = h[i - 1] / 6.
+            b[i - 1, i] = h[i - 1] / 6.
+
+            d[i, i] = -1. / h[i - 1] - 1. / h[i]
+            d[i, i - 1] = 1. / h[i - 1]
+            d[i - 1, i] = 1. / h[i - 1]
+
+        return b, d
+
+    def _get_s(self, b, d):
+        return d.T.dot(np.linalg.inv(b)).dot(d)
+
+
+class CyclicCubicSplines(MultivariateGamSmoother):
+    def __init__(self, x, df, variables_name=None):
+        self.dfs = df
+        super(CyclicCubicSplines, self).__init__(x, variables_name)
+        return
+
+    def _make_smoothers_list(self):
+        smoothers = []
+        for v in range(self.k_variables):
+            smoothers.append(UnivariateCubicCyclicSplines(self.x[:, v], df=self.dfs[v],
+                                                          variable_name=self.variables_name[v]))
+
+        return smoothers
+
+# class CubicRegressionSplines(BaseCubicSplines):
+#     # TODO: this class is still not tested
+#
+#     def __init__(self, x, df=10):
+#         import warnings
+#         warnings.warn("This class is still not tested and it is probably not working properly. "
+#                       "I suggest to use another smoother", Warning)
+#
+#         super().__init__(x, df)
```

Comment:
fix `super(CubicRegressionSplines, self)` even though it's currently commented out